hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java

```diff
@@ -2712,10 +2713,11 @@ public GrantResponse grant(RpcController controller, GrantRequest request)
       if (master.cpHost != null) {
         master.cpHost.preGrant(perm, mergeExistingPermissions);
       }
-      try (Table table = master.getConnection().getTable(PermissionStorage.ACL_TABLE_NAME)) {
-        PermissionStorage.addUserPermission(getConfiguration(), perm, table,
-          mergeExistingPermissions);
-      }
+      ProcedurePrepareLatch latch = ProcedurePrepareLatch.createBlockingLatch();
```

Comment:
Maybe a better way is to add a proc id field in GrantResponse, and query the result at client side?